content/YubiHSM2/Usage_Guides/YubiHSM_2_for_Microsoft_SQL_Server_Deployment_Guide--Enabling_Always_Encrypted_with_YubiHSM_2/Use_PowerShell_Script_to_Generate_the_CMK_and_CEK.adoc

```diff
@@ -0,0 +1,115 @@
+
+
+=== Create a CMK in the YubiHSM 2 with CNG Provider (KSP)
+
+....
+$cngProviderName = "YubiHSM Key Storage Provider"
+$cngAlgorithmName = "RSA"
+$cngKeySize = 2048 # Recommended key size for column master keys
+$cngKeyName = "AlwaysEncryptedKey-PS" # Name identifying your key in the KSP
+$cngProvider = New-Object System.Security.Cryptography.CngProvider($cngProviderName)
+$cngKeyParams = New-Object System.Security.Cryptography.CngKeyCreationParameters
+$cngKeyParams.provider = $cngProvider
+$cngKeyParams.KeyCreationOptions = [System.Security.Cryptography.CngKeyCreationOptions]::OverwriteExistingKey
+$keySizeProperty = New-Object System.Security.Cryptography.CngProperty("Length", [System.BitConverter]::GetBytes($cngKeySize), [System.Security.Cryptography.CngPropertyOptions]::None);
+$cngKeyParams.Parameters.Add($keySizeProperty)
+$cngAlgorithm = New-Object System.Security.Cryptography.CngAlgorithm($cngAlgorithmName)
+$cngKey = [System.Security.Cryptography.CngKey]::Create($cngAlgorithm, $cngKeyName, $cngKeyParams)
+....
+
+
+=== Import SQL Server Module
+
+....
+Import-Module "SqlServer"
+....
+
+
+=== Connect to your Database
+
+....
+$serverName = "<server name>"
+$databaseName = "<database name>"
+$connStr = "Server = " + $serverName + "; Database = " + $databaseName + "; Integrated Security = True"
+$connection = New-Object Microsoft.SqlServer.Management.Common.ServerConnection
+$connection.ConnectionString = $connStr
+$connection.Connect()
+$server = New-Object Microsoft.SqlServer.Management.Smo.Server($connection)
+$database = $server.Databases[$databaseName]
+....
+
+
+=== Create SQL CMK Settings Object for your CMK
+
+....
+$cmkSettings = New-SqlCngColumnMasterKeySettings -CngProviderName $cngProviderName -KeyName $cngKeyName
+....
+
+
+=== Create CMK Metadata in Database
+
+....
+$cmkName = "CMK-YubiHSM-PS"
+New-SqlColumnMasterKey -Name $cmkName -InputObject $database -ColumnMasterKeySettings $cmkSettings -Verbose
+....
+
+
+=== Generate CEK, Encrypt with CMK, and Create CEK Metadata in Database
+
+....
+$cekName = "CEK-YubiHSM-PS"
+New-SqlColumnEncryptionKey -Name $cekName -InputObject $database -ColumnMasterKeyName $cmkName -Verbose
+....
+
+
+=== Customize the Script
+
+*Step 1* To customize this script, change the placeholders `server name` and `database name` to the actual values of the Microsoft SQL Server name and the database used. For the test database used in this example, the `database name` is set to `Sales3`, while the `server name` should be set to the name of your Windows server.
+
+*Step 2* Save the PowerShell script file in a folder with an appropriate name, for example `CreateColumnMasterAndEncryptionKeys- YubiHSM.ps1`.
```

Comment:
"CreateColumnMasterAndEncryptionKeys- YubiHSM.ps1" has a space here, that should probably not be there?

Reply:
This is what it was in the Google doc: 
"CreateColumnMasterAndEncryptionKeys- YubiHSM.ps1." but I am removing the space. 
8c0cdada85a7d40226ba4dbb4ac48dfdd8cf5be3